File: UCloudRtcSdk-ios-demo/UCloudRtcSdk-ios-demo/urtc/MeetingRoomCell.m
//
//  MeetingRoomCell.m
//  MeetingSDK
//
//  Created by tony on 2019/4/28.
//  Copyright © 2018年 ucloud. All rights reserved.
//

#import "MeetingRoomCell.h"



@interface MeetingRoomCell ()
@property (nonatomic, weak) UCloudRtcStream *stream;
@property (weak, nonatomic) IBOutlet UILabel *streamLabel;
@end

@implementation MeetingRoomCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
    
    self.backgroundColor = [UIColor grayColor];
}


- (void)configureWithStream:(UCloudRtcStream *)stream {
    self.stream = stream;
    
    [stream renderOnView:self.contentView];
    self.streamLabel.text = [NSString stringWithFormat:@"id:%@",stream.streamId];
    [self.contentView bringSubviewToFront:self.streamLabel];
    
    UIButton *audioBtn = [[UIButton alloc] initWithFrame:CGRectMake(self.frame.size.width - 20, self.frame.size.height - 20, 20, 20)];
    [audioBtn setImage:[UIImage imageNamed:@"microphone.png"] forState:UIControlStateNormal];
    [audioBtn setImage:[UIImage imageNamed:@"microphone_close.png"] forState:UIControlStateSelected];
    [audioBtn addTarget:self action:@selector(audioClick:) forControlEvents:UIControlEventTouchUpInside];
    [self.contentView addSubview:audioBtn];
    
    
    UIButton *videoBtn = [[UIButton alloc] initWithFrame:CGRectMake(self.frame.size.width - 45, self.frame.size.height - 20, 20, 20)];
    [videoBtn setImage:[UIImage imageNamed:@"camera_btn_on.png"] forState:UIControlStateNormal];
    [videoBtn setImage:[UIImage imageNamed:@"camera_btn_off.png"] forState:UIControlStateSelected];
    [videoBtn addTarget:self action:@selector(videoClick:) forControlEvents:UIControlEventTouchUpInside];
    [self.contentView addSubview:videoBtn];
}

- (void)layoutSubviews {
    [super layoutSubviews];
//    self.stream.preview.frame = self.contentView.bounds;
}

-(void)audioClick:(UIButton *)btn {
    btn.selected =!btn.selected;
    if (self.delegate && [self.delegate respondsToSelector:@selector(didMuteStream:muteAudio:)]) {
        [self.delegate didMuteStream:self.stream muteAudio:btn.selected];
    }
}

-(void)videoClick:(UIButton *)btn {
    btn.selected =!btn.selected;
    if (self.delegate && [self.delegate respondsToSelector:@selector(didMuteStream:muteVideo:)]) {
        [self.delegate didMuteStream:self.stream muteVideo:btn.selected];
    }
}
@end

File: Example/UCloudRtcSdk-ios-demo-swift/UCloudRtcSdk-ios-demo-swift/MeetingRoomCell.swift
//
//  MeetingRoomCell.m
//  UCloudRtcSdkDemo
//
//  Created by tony on 2019/4/28.
//  Copyright © 2019年 ucloud. All rights reserved.
//

protocol MeetingRoomCellDelegate:NSObjectProtocol{
    /**开启或关闭视频*/
    func didMuteStream(stream:UCloudRtcStream,muteVideo:Bool)
    /**开启或关闭音频*/
    func didMuteStream(stream:UCloudRtcStream,muteAudio:Bool)
}

class MeetingRoomCell:UICollectionViewCell {
    
    public var stream : UCloudRtcStream?
    public var delegate : MeetingRoomCellDelegate?
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.backgroundColor = UIColor.gray
    }
    
    func configureWithStream(stream:UCloudRtcStream) {
        self.stream = stream
        stream.render(on: self.contentView)
        if (stream.userId != nil) {
            let streamLabel: UILabel = UILabel(frame:CGRect(x:0, y:0, width:self.frame.size.width, height:30))
            streamLabel.textColor = UIColor.white
            streamLabel.textAlignment = .center
            streamLabel.text = String(format: "id: %@", stream.userId)
            self.contentView.addSubview(streamLabel)
            let audioBtn: UIButton = UIButton(frame: CGRect(x:self.frame.size.width-20, y:self.frame.size.height-20, width:20, height:20))
            audioBtn.setImage(UIImage(named: "microphone.png"),for: .normal)
            audioBtn.setImage(UIImage(named: "microphone_close.png"), for: .selected)
            audioBtn.addTarget(self, action: Selector("audioClick:"), for: .touchUpInside)
            self.contentView.addSubview(audioBtn)
            let videoBtn: UIButton = UIButton(frame: CGRect(x:self.frame.size.width-45, y:self.frame.size.height-20, width:20, height:20))
            videoBtn.setImage(UIImage(named: "camera_btn_on.png"), for: .normal)
            videoBtn.setImage(UIImage(named: "camera_btn_off.png"), for: .selected)
            videoBtn.addTarget(self, action: Selector("videoClick:"), for: .touchUpInside)
            self.contentView.addSubview(videoBtn)
        }
    }

    
   @objc func audioClick(_ btn: UIButton) {
        btn.isSelected = !btn.isSelected
        self.delegate?.didMuteStream(stream: self.stream!, muteAudio: btn.isSelected)
    }
    
   @objc func videoClick(_ btn: UIButton) {
        btn.isSelected = !btn.isSelected
        self.delegate?.didMuteStream(stream: self.stream!, muteVideo: btn.isSelected)
    }
    
}

